src/mmw/js/src/modeling/views.js

```diff
@@ -130,6 +177,64 @@ var ScenarioTabPanelView = Marionette.ItemView.extend({
 
     updateSlug: function() {
         this.model.slugifyName();
+        this.render();
+    },
+    ui: {
+        share: '.share',
+        destroyConfirm: '.delete',
+        rename: '.rename',
+        print: '.print' 
+    },
+    events: {
+        'click @ui.rename': 'renameScenario',
+        'click @ui.destroyConfirm': 'destroyConfirm',
+        'click @ui.print': function() {
+            window.print();
+        },
+        'click @ui.share': 'showShareModal'
+    },
+    renameScenario: function() {
+        var self = this,
+            $nameField = this.$el.find('.tab-name');
+
+        $nameField.attr('contenteditable', true).focus();
+
+        $nameField.on('keyup', function(e) {
+            // Cancel on escape key.
+            if (e.keyCode === ESCAPE_KEYCODE) {
+                self.render();
+            }
+        });
+
+        $nameField.on('keypress', function(e) {
+            var keycode = (e.keyCode ? e.keyCode : e.which);
+            if (keycode == ENTER_KEYCODE) {
+                // Don't add line returns to the text.
+                e.preventDefault();
+                self.triggerMethod('rename:scenario', self.model, $(this).text());
+            }
+        });
+
+        $nameField.on('blur', function(e) {
+           self.triggerMethod('rename:scenario', self.model, $(this).text());
+        });
+    },
+
+    destroyConfirm: function() {
+        var del = new coreViews.DeleteModal({
+            objToDelete: this.model,
+            model: new Backbone.Model({ deleteLabel: 'scenario' })
+        });
+        del.render();
+        del.$el.modal('show');
+    },
+
+    showShareModal: function() {
+        var share = new coreViews.ShareModal({
+            model: new Backbone.Model({ url: window.location.href })
```

Comment:
If we're generating a link to the active scenario, that slug should be in the URL.  However, if this model isn't persisted yet (we're not logged in) we don't have a URL to generate so we should prevent this behavior.